### DISCUSSION TITLE
controlling z-index

### DISCUSSION THREAD

**Discussion by dimadem** [P5-wrapper/react#221] [Q&A]
Hello every one, trying to set z-index in project (in case wanna have p5sketch as a part of background). 

Is there any rules to use it with component ?

thank you!

**Comment by jamesrweb** [comment]
As mentioned in [the Children section](https://github.com/P5-wrapper/react/blob/master/README.md#children) of the project readme, you can access the wrapper class and then use any css-in-js implementation you like therein! 👌🏻

**Reply by dimadem** [reply]
Thanks a lot! Don't know about that. =)

**Comment by dimadem** [comment]
<img width="467" alt="Снимок экрана 2023-01-27 в 21 40 32" src="https://user-images.githubusercontent.com/50800563/215205815-b745a32b-3645-4eba-9698-af3ee276865e.png">

`  return (
    <P5WrapperClassName className="absolute">
      <ReactP5Wrapper sketch={sketch} />
    </P5WrapperClassName>
  );`

Is that okay case for Tailwind? Trying to control "z-index" or "absolute" and its not able for me. Sorry I'm not a great performer and don't have deep knowledge. So what can i do? Thank you

In case any tailwind property is not able there

P.S. I can’t understand concept behind P5WrapperClassName. It’s some kind of hook? 

🎈🎈🎈

Using 
last version of react-p5-wrapper
tailwindui
nextjs

**Reply by jamesrweb** [reply]
`P5WrapperClassName` is the class name given to the `div` that wraps the `canvas` element.

In your case you would want to target `${P5WrapperClassName}` and `${P5WrapperClassName} canvas` if you were using a css-in-js library. To use tailwind, you would want to actually go to your `.css` file where you `@import tailwind` etc and then add a style there for `react-p5-wrapper div` (the wrapping div) and `react-p5-wrapper div canvas` (the canvas) and then use `@apply some-tailwind-class-name` in those selector blocks accordingly.

That is all tailwind specific though and out of the scope of this project, all you need to know is that the `P5WrapperClassName` is the class name of the `div` element that wraps the `canvas` element.

**Comment by dimadem** [comment]
Another try 

```
import { ReactP5Wrapper, P5WrapperClassName } from "react-p5-wrapper";
import sketch from "../mikado/sketch";

//TODO get rid of the z-index */
//TODO get transparent background */

export default function P5Background(children) {
  const ZindexProperty = () => {
    return (
      <style jsx global>
        {`
          .${P5WrapperClassName} {
           relative, -z-10
          }
        `}
      </style>
    );
  };
  return (
    <>
      <ZindexProperty />
      <ReactP5Wrapper className="relative -z-10" sketch={sketch} />
    </>
  );
}
```

<img width="464" alt="Снимок экрана 2023-01-28 в 10 57 51" src="https://user-images.githubusercontent.com/50800563/215262926-9b79f0f8-917b-420d-8157-c3a2d2463ba7.png">

**Reply by jamesrweb** [reply]
Yes but:

```css
relative, -z-10
```

is not valid css and will not work outside of the css file you use to import tailwind itself without using `@apply` rules. Which again, is tailwind specific and out of scope for this project, please read the tailwind docs for more on this.
